src/FilterButtonList.js

```diff
@@ -50,57 +61,99 @@ let FilterButtonItem = _.flow(
             </div>
-            <Button onClick={() => tree.clear(node.path)}>Clear</Button>
-            <Button primary onClick={F.off(modal)}>
-              Done
-            </Button>
-          </div>
+            <Flex style={{ justifyContent: 'flex-end' }}>
+              <Button onClick={() => tree.clear(node.path)}>
+                Clear Filter
+              </Button>
+              <Button
+                primary
+                onClick={() => {
+                  F.off(modal)()
+                  tree.mutate(node.path, { paused: true })
+                }}
+                style={{ marginLeft: '10px' }}
+              >
+                Done
+              </Button>
+            </Flex>
+          </Flex>
         </Modal>
-      </>
+      </div>
     )
   }
 )
 
-let GroupBox = ({ nodeJoinColor, children, nested, className }) => (
+let GroupBox = ({ nodeJoinColor, children, nested, className, style }) => (
   <Flex
     wrap
     className={`${className} ${nested ? 'nested' : ''}`}
     alignItems="center"
-    style={{ borderColor: nodeJoinColor }}
+    style={{ borderColor: nodeJoinColor, ...style }}
   >
     {children}
   </Flex>
 )
 
-let FilterButtonList = ({
-  node,
-  tree,
-  fields = {},
-  mapNodeToProps = _.noop,
-  className = 'filter-button-list',
-  nested = false,
-}) => (
-  <GroupBox
-    {...{ nested, className }}
-    nodeJoinColor={node && styles.joinColor(node)}
-  >
-    {_.map(child => {
-      let Component = child.children ? FilterButtonList : FilterButtonItem
-      return (
-        <Component
-          key={child.path}
-          nested
-          {...{
-            tree,
-            node: child,
-            fields,
-            mapNodeToProps,
-            className,
-          }}
-        />
+let FilterButtonList = observer(
+  ({
+    node,
+    tree,
+    fields = {},
+    mapNodeToProps = _.noop,
+    allowDuplicateFields = false,
+    className = '',
+    nested = false,
+    style,
+    children,
+    theme: { Icon, Button, FilterButton = Button },
+  }) => {
+    let options = fieldsToOptions(fields)
+    if (!allowDuplicateFields) {
+      options = _.reject(
+        x => _.includes(x.field, getGroupFields(node)),
+        options
       )
-    }, _.get('children', node))}
-  </GroupBox>
+    }
+    return (
+      <GroupBox
+        className={`filter-button-list ${className}`}
+        {...{ nested, style }}
+        nodeJoinColor={node && styles.joinColor(node)}
+      >
+        {children}
+        {_.map(child => {
+          let Component = child.children ? FilterButtonList : FilterButtonItem
+          return (
+            <Component
+              key={child.path}
+              nested
+              {...{
+                tree,
+                node: child,
+                fields,
+                mapNodeToProps,
+                className,
+              }}
+            />
+          )
+        }, _.get('children', node))}
+
+        {!nested && (
+          <div>
```

Comment:
I’m pretty sure we already have cases where we have a filter adder (either in storybooks or in spark). Did you consider those existing cases? We’ll need to update them if filter button list packages up a filter adder

Reply:
Yeah, I've added an optional `addFilters` prop